hazelcast/src/main/java/com/hazelcast/instance/impl/DefaultNodeExtension.java

```diff
@@ -283,63 +288,85 @@ private void printNodeInfoInternal(BuildInfo buildInfo, String build) {
     private void printSecurityFeaturesInfo(Config config) {
-        StringBuilder sb = new StringBuilder("\n🔒Status of security related features:");
-        addFeatureInfo(sb, "Custom cluster name used", !Config.DEFAULT_CLUSTER_NAME.equals(config.getClusterName()));
+        StringBuilder sb = new StringBuilder("\n🔒Security recommendations and their status:");
+        addSecurityFeatureCheck(sb, "Use a custom cluster name", !Config.DEFAULT_CLUSTER_NAME.equals(config.getClusterName()));
         boolean isMulticastJoin = node.shouldUseMulticastJoiner(getActiveMemberNetworkConfig(config).getJoin());
-        addFeatureInfo(sb, "Member multicast discovery/join method disabled", !isMulticastJoin);
-        addFeatureInfo(sb, "Advanced networking used - Sockets separated", config.getAdvancedNetworkConfig().isEnabled());
+        addSecurityFeatureCheck(sb, "Disable member multicast discovery/join method", !isMulticastJoin);
+
         AdvancedNetworkConfig advancedNetworkConfig = config.getAdvancedNetworkConfig();
-        addFeatureInfo(sb, "Server sockets bound to a single network interface",
-                !node.getProperties().getBoolean(ClusterProperty.SOCKET_SERVER_BIND_ANY));
-        AuditlogConfig auditlogConfig = config.getAuditlogConfig();
+        addSecurityFeatureCheck(sb, "Use advanced networking, separate client and member sockets",
+                advancedNetworkConfig.isEnabled());
+        boolean bindAny = node.getProperties().getBoolean(SOCKET_SERVER_BIND_ANY);
+        addSecurityFeatureCheck(sb,
+                "Bind Server sockets to a single network interface (disable " + SOCKET_SERVER_BIND_ANY.getName() + ")",
+                !bindAny);
+        StringBuilder tlsSb = new StringBuilder();
+        boolean tlsUsed = true;
+        if (advancedNetworkConfig.isEnabled()) {
+            for (Map.Entry<EndpointQualifier, EndpointConfig> e : advancedNetworkConfig.getEndpointConfigs().entrySet()) {
+                tlsUsed = addAdvNetworkTlsInfo(tlsSb, e.getKey(), e.getValue().getSSLConfig()) && tlsUsed;
+            }
+        } else {
+            SSLConfig sslConfig = config.getNetworkConfig().getSSLConfig();
+            tlsUsed = addSecurityFeatureCheck(tlsSb, "Use TLS communication protection (Enterprise)",
+                    sslConfig != null && sslConfig.isEnabled());
+        }
         boolean jetEnabled = config.getJetConfig().isEnabled();
-        addFeatureInfo(sb, "Jet disabled", !jetEnabled);
         if (jetEnabled) {
-            addFeatureInfo(sb, "Jet resource upload disabled", !config.getJetConfig().isResourceUploadEnabled());
+            boolean trustedEnv = tlsUsed || !bindAny;
+            addSecurityFeatureCheck(sb, "Use Jet in trusted environments only (single network interface and/or TLS enabled)",
+                    trustedEnv);
+            addSecurityFeatureCheck(sb, "Disable Jet resource upload", !config.getJetConfig().isResourceUploadEnabled());
```

Comment:
this should use the same wording as above in terms of trusted environments and not ask the user to disable resource upload

Reply:
I was considering it. Then it would be always green, when the Jet itself is green. So the message would be superfluous. And it is not, IMO. The resource uploads come with greater risks.

An alternative could be replacing this warning with info message. Something like "Jet resource upload is enabled. Any uploaded code can be executed within the Hazelcast. Use this in trusted environments only."
@cangencer WDYT?